### DISCUSSION TITLE
Share code for simulator configure in Snapshot lane with Scan lane

### DISCUSSION THREAD

**Discussion by cooler333** [fastlane/fastlane#17576] [Ideas]
For Example:

https://github.com/fastlane/fastlane/blob/5ff0ac2fa027447b90a73c744a19cb3a29e4ef3b/snapshot/lib/snapshot/simulator_launchers/simulator_launcher_base.rb#L170 - Dark/Light theme

https://github.com/fastlane/fastlane/blob/5ff0ac2fa027447b90a73c744a19cb3a29e4ef3b/snapshot/lib/snapshot/simulator_launchers/simulator_launcher_base.rb#L158 - Languages

Use case:
we use Scan (unit/ui tests) to make snapshot comparison (reference/current) and now we copypaste some scripts or write our own.

**Comment by rogerluan** [comment]
Hi @cooler333 👋 

Your request is not clear to me, what do you mean exactly? Can you give more details of what you're trying to achieve?

**Comment by artFintch** [comment]
Hey @rogerluan 👋🏻

We use `Scan` lane for snapshots UI testing. And we wonder if we can reuse some private methods from `Snapshot` lane. It would help us to prepare simulators.

For example, change locale in `Scan` lane as it happening in `Snapshot` lane:
https://github.com/fastlane/fastlane/blob/5ff0ac2fa027447b90a73c744a19cb3a29e4ef3b/snapshot/lib/snapshot/simulator_launchers/simulator_launcher_base.rb#L150
Set dark mode, etc:
https://github.com/fastlane/fastlane/blob/5ff0ac2fa027447b90a73c744a19cb3a29e4ef3b/snapshot/lib/snapshot/simulator_launchers/simulator_launcher_base.rb#L164
<hr>

As result, It will be great to have the ability to pass similar arguments for `Scan` lane like `languages: ["ru"]`.
What do you think about this @rogerluan? cc @cooler333

**Reply by rogerluan** [reply]
@artFintch @cooler333 Humm, if I understand correctly, you're looking to pass extra arguments to Snapshot Simulator launcher. Have you looked into [all snapshot options](https://docs.fastlane.tools/actions/snapshot/), such as `launch_arguments`, `languages`, `localize_simulator`, `dark_mode`, etc? Looks like that would help you achieve what you're looking for.

Let me know if I understood something different, so I can further assists 😊

**Reply by swiftyfinch** [reply]
@rogerluan Not really 😀 But thanks for your answer!
We don't use `Snapshot` at all, but we use `Scan`. And we are looking for the official ability to use `Snapshot` options for `Scan`.

**Reply by rogerluan** [reply]
Oh, got it! That makes sense.
Unfortunately I _think_ there's no built-in way to do that, in fact. It's a nice suggestion for a new functionality, I definitely see value in it 👍

**Reply by cooler333** [reply]
> nice suggestion for a new functionality

how can I do that?

**Reply by rogerluan** [reply]
@cooler333 you can start by checking [CONTRIBUTING.md](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md) and diving into the code 🤗 Related files are probably `scan/lib/scan/options.rb` and `snapshot/lib/snapshot/options.rb`
